### PR TITLE
Add basic testing setup with mocha

### PR DESCRIPTION
**Pull request by domharrington**
Tests are added by creating a new folder inside of test/ and adding 2 files to the directory:

- swagger.json - input swagger file
- oas.json - expected output oas file

The test runner at test/index.test.js iterates through these and performs a deepEqual on the output from running it through swagger2openapi.

I had to remove `test/` from the gitignore, i assume this is being used somewhere by your `testRunner` script? This may clash with that, so feel free to take this and work from it or let me know what changes need to happen

### CODE REVIEW

**Comment by MikeRalphson** [comment]
Thank you!

* The `.gitignore`d `/test` directory was only used by old manual tests, so that's not a problem.
* bikeshed: could we use `openapi.json` instead of `oas.json`?
* `.travis.yml` should be modified to run the new tests
* use of `async`/`await` - not a showstopper, although Node 4 is still supported for the converter itself. Travis should run the new tests only where `async`/`await` support is present (test for Node >= 8.x would be fine)

Wondering whether it would be preferable to read the testcases with `js-yaml` (as json is a subset of yaml, but the reverse is not true) and have them named `.yaml`?

I'm happy to make any or all of those changes.

Thanks again.

**Comment by domharrington** [comment]
Did not actually know that they preferred the file to be called `openapi.json`. Just read that part of the spec now: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#document-structure

I've made the other changes and removed async/await from the tests so that it can still be tested against node 4.

File: test/index.test.js [hunk 1]
```diff
@@ -10,11 +10,17 @@ const tests = fs.readdirSync(__dirname).filter(file => {
 
 tests.forEach(async (test) => {
```
Comment (MikeRalphson) [inline test/index.test.js]: Can we remove the `async` directive from here now? CI test failed on Node 4/6.

Comment (domharrington) [reply test/index.test.js]: Whooops. Thought I removed them all. Dont need that one!

Comment (domharrington) [reply test/index.test.js]: It's a shame the tests take so long to run!

File: .travis.yml [hunk 2]
```diff
@@ -15,4 +15,4 @@ script:
 #  - travis_wait node $nflags testRunner -q hapi/APIs/
   - travis_wait node $nflags testRunner -q -n openapi3-examples/pass
   - travis_wait node $nflags testRunner -q -r -n -f openapi3-examples/fail
-
+  - npm test
```
Comment (MikeRalphson) [inline .travis.yml]: Ah, for node 4, we're going to need something like

`node node_modules/mocha/bin/mocha $nflags`

Sorry.

**Comment by MikeRalphson** [comment]
LGTM. Thanks again.

**Pull request merged**